### PR TITLE
Check Title Length

### PR DESCRIPTION
**Pull request by csgillespie**

### CODE REVIEW

**Comment by csgillespie** [comment]
Two changes

1. Condense the NEWS.md in 0.X.*
2. Add a quick check for title length. If the title is longer 25 ish characters it goes into the border of the front cover

File: NEWS.md [hunk 1]
```diff
@@ -126,125 +80,56 @@
...
-# 0.3.4
   * Bug in create_final due to re-arranging of gitlab repos
-
-# 0.3.2
   * Copy logo; fixes annoying bug involving spaces and directory names.
-
-# 0.3.1
   * Normalise path for logo
   
 # 0.1.9
```
Comment (jwalton3141) [inline NEWS.md:127]: # 0.1.*

Comment (jwalton3141) [inline NEWS.md:127]: Collapse to 0.1*

**Comment by jwalton3141** [comment]
I like collapsing NEWS.md as we start a new major version; sensible

File: NEWS.md (hunk 1 above)
Comment (csgillespie) [reply NEWS.md]: I forgot to collapse 0.1.9 to 0.1.*

**Comment by jwalton3141** [comment]
@csgillespie Okay, so @JohnMcIntyreJR showed me a case in which TeX broke the title by hyphenating a word, so I see why adding a manual linebreak _may_ be necessary in some cases.

I still wonder whether a TeX solution might be better than another check? Ie. something like in this https://tex.stackexchange.com/questions/5001/prevent-hyphenation-and-use-line-break-in-block-of-text

(Change the titlepage.tex to stop words from hyphenating at linebreaks)

**Comment by csgillespie** [comment]
So I'm not opposed to a TeX type solution.

But I think we should always explicitly state where the line break
should be. The title is quite important, and actively thinking about breaks
isn't a bad idea. For example, how many lines would we want a title to
cover?


Dr Colin Gillespie
http://www.mas.ncl.ac.uk/~ncsg3/

**Comment by jwalton3141** [comment]
Okay, I think I've come around to the idea

I think I'll make a PR for a change to titlepage.tex to prevent linebreaking within words anyway.

But, when choosing line breaks we also need to consider the content and structure of the title, which we need to think about manually anyway.

**Review by jwalton3141 — APPROVED** [review]

File: R/check_config.R [hunk 2]
```diff
@@ -9,6 +9,18 @@
 check_config = function() {
   msg_start("Checking config file...check_config()")
   config_issue = FALSE
+
+  ## Check length of title on front page
```
[the remark this answers is not in the document]
Comment (csgillespie) [reply R/check_config.R:13]: But what about `front: "Introduction to Bayesiannn inference using RStan"`?

[the remark this answers is not in the document]
Comment (jwalton3141) [reply R/check_config.R:13]: I'm not sure I understand this comment. Would the typo cause hyphenation in the middle of the word? Doesn't matter if so as I've come around to your manual line-break philosophy (I made the above comment on the RStan course last night, even though github seems to be listing it as ~45 mins ago)

[the remark this answers is not in the document]
Comment (csgillespie) [reply R/check_config.R:13]: @jwalton3141 Sorry, what I meant was you could have a title that latex would add a line break, but would still be too close to the margin. The extra `nn` was just padding

**Pull request merged**